src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -16,25 +16,36 @@ internal abstract partial class ConversionsBase
         private const int MaximumRecursionDepth = 50;
 
         protected readonly AssemblySymbol corLibrary;
+        private readonly ConversionsBase _withoutNullability;
         private readonly int _currentRecursionDepth;
 
-        protected ConversionsBase(AssemblySymbol corLibrary, int currentRecursionDepth)
+        protected ConversionsBase(AssemblySymbol corLibrary, int currentRecursionDepth, ConversionsBase withoutNullability)
         {
             Debug.Assert((object)corLibrary != null);
+            Debug.Assert(withoutNullability == null || !withoutNullability.IncludeNullability);
+
             this.corLibrary = corLibrary;
+            _withoutNullability = withoutNullability;
             _currentRecursionDepth = currentRecursionDepth;
         }
 
         public abstract Conversion GetMethodGroupConversion(BoundMethodGroup source, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics);
 
         public abstract Conversion GetStackAllocConversion(BoundStackAllocArrayCreation sourceExpression, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics);
 
-        protected abstract ConversionsBase CreateInstance(int currentRecursionDepth);
+        protected abstract ConversionsBase CreateInstance(int currentRecursionDepth, ConversionsBase withoutNullability);
 
         protected abstract Conversion GetInterpolatedStringConversion(BoundInterpolatedString source, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics);
 
         internal AssemblySymbol CorLibrary { get { return corLibrary; } }
 
+        internal bool IncludeNullability => _withoutNullability != null;
+
+        internal ConversionsBase WithoutNullability()
```

Comment:
This method doesn't seem to be referenced. If removed, `_withoutNullability` can be replaced with a `bool` field.

Reply:
Removed.